src/univariate/locationscale.jl

```diff
@@ -106,36 +114,52 @@ mode(d::AffineDistribution) = d.μ + d.σ * mode(d.ρ)
 modes(d::AffineDistribution) = d.μ .+ d.σ .* modes(d.ρ)
 
 var(d::AffineDistribution) = d.σ^2 * var(d.ρ)
-std(d::AffineDistribution) = d.σ * std(d.ρ)
-skewness(d::AffineDistribution) = skewness(d.ρ)
+std(d::AffineDistribution) = abs(d.σ) * std(d.ρ)
+skewness(d::AffineDistribution) = sign(d.σ) * skewness(d.ρ)
 kurtosis(d::AffineDistribution) = kurtosis(d.ρ)
 
 isplatykurtic(d::AffineDistribution) = isplatykurtic(d.ρ)
 isleptokurtic(d::AffineDistribution) = isleptokurtic(d.ρ)
 ismesokurtic(d::AffineDistribution) = ismesokurtic(d.ρ)
 
-entropy(d::ContinuousAffineDistribution) = entropy(d.ρ) + log(d.σ)
+entropy(d::ContinuousAffineDistribution) = entropy(d.ρ) + log(abs(d.σ))
 entropy(d::DiscreteAffineDistribution) = entropy(d.ρ)
 
 mgf(d::AffineDistribution,t::Real) = exp(d.μ*t) * mgf(d.ρ,d.σ*t)
+cf(d::AffineDistribution, t::Real) = cf(d.ρ,t*d.σ) * exp(1im*t*d.μ)
 
 #### Evaluation & Sampling
 
-pdf(d::ContinuousAffineDistribution, x::Real) = pdf(d.ρ,(x-d.μ)/d.σ) / d.σ
+pdf(d::ContinuousAffineDistribution, x::Real) = pdf(d.ρ,(x-d.μ)/d.σ) / abs(d.σ)
 pdf(d::DiscreteAffineDistribution, x::Real) = pdf(d.ρ,(x-d.μ)/d.σ)
 
-logpdf(d::ContinuousAffineDistribution,x::Real) = logpdf(d.ρ,(x-d.μ)/d.σ) - log(d.σ)
+logpdf(d::ContinuousAffineDistribution,x::Real) = logpdf(d.ρ,(x-d.μ)/d.σ) - log(abs(d.σ))
 logpdf(d::DiscreteAffineDistribution, x::Real) = logpdf(d.ρ,(x-d.μ)/d.σ)
 
-for f in (:cdf, :ccdf, :logcdf, :logccdf)
-    @eval $f(d::AffineDistribution, x::Real) = $f(d.ρ, (x - d.μ) / d.σ)
+# CDF methods
+
+const CDF_FNS = (:cdf, :logcdf, :logccdf, :ccdf)
+for (f, fc) in zip(CDF_FNS, reverse(CDF_FNS))
+    @eval function $f(d::ContinuousAffineDistribution, x::Real)
+        x = (x - d.μ) / d.σ
+        return d.σ > 0 ? $f(d.ρ, x) : $fc(d.ρ, x)
+    end
 end
 
-quantile(d::AffineDistribution,q::Real) = d.μ + d.σ * quantile(d.ρ,q)
+function cdf(d::DiscreteAffineDistribution, x::Real)
+    x = (x - d.μ) / d.σ
+    # Have to include probability mass at endpoints
+    return d.σ > 0 ? cdf(d.ρ, x) : (ccdf(d.ρ, x) + pdf(d.ρ, x))
+end
+ccdf(d::DiscreteAffineDistribution, x::Real) = 1 - cdf(d, x)
```

Comment:
This is different from the current definitions - currently we do not use the default 1 - cdf.
```suggestion
function ccdf(d::DiscreteAffineDistribution, x::Real)
    z = (x - d.μ) / d.σ
    # Have to exlude probability mass at endpoints
    return d.σ > 0 ? ccdf(d.ρ, z) : (cdf(d.ρ, z) - pdf(d.ρ, x))
end
```